### PR TITLE
Feat/swarinfo 488 vm reset button

### PR DESCRIPTION
**Pull request by xgroleau**
VM resets with the button 1

### CODE REVIEW

File: src/Main.cpp [hunk 1]
```diff
@@ -68,39 +82,50 @@ class BittyBuzzTask : public AbstractTask<10 * configMINIMAL_STACK_SIZE> {
...
-                break;
-            }
-            case BBVMRet::VmErr: {
-                m_logger.log(LogLevel::Error, "BBZVM failed to step. state: %s err: %s",
+            m_deviceStateUI.setDeviceState(DeviceState::Ok);
+
+            if (!m_bittybuzzVm.init(buzzLibraries.data(), buzzLibraries.size()) ||
+                !m_bittybuzzVm.start()) {
+
+                m_logger.log(LogLevel::Error, "",
```
Comment (cquesnel) [inline src/Main.cpp:94]: Not logging anything here?

File: src/Main.cpp [hunk 2]
```diff
@@ -68,39 +82,50 @@ class BittyBuzzTask : public AbstractTask<10 * configMINIMAL_STACK_SIZE> {
...
+                }
+                case BBVMRet::VmErr: {
+                    m_logger.log(LogLevel::Error, "BBZVM failed to step. state: %s err: %s",
+                                 BittyBuzzSystem::getStateString(m_bittybuzzVm.getState()),
+                                 BittyBuzzSystem::getErrorString(m_bittybuzzVm.getError()));
+                    m_deviceStateUI.setDeviceState(vmErrorToDeviceState(m_bittybuzzVm.getError()));
+                    break;
+                }
+                }
+                Task::delay(100);
```
Comment (cquesnel) [inline src/Main.cpp:124]: Should this be in a constant somewhere? It seems like something we might want to let the user configure

File: src/Main.cpp (hunk 1 above)
Comment (xgroleau) [reply src/Main.cpp:94]: idk why the format was removed, fixed

File: src/Main.cpp (hunk 2 above)
Comment (xgroleau) [reply src/Main.cpp:124]: It's in the next PR don't worry. And the VM will stop printing endlessy when it cashes

**Pull request merged**